axlearn/common/base_layer.py

```diff
@@ -296,6 +365,8 @@ def initialize_parameters_recursively(
                     parameter_spec=spec,
                 )
         for name, child in self._children.items():
+            if not isinstance(child, BaseLayer):
+                continue
```

Comment:
Could you add a comment about why this is needed and a test that fails without this change?

Reply:
Added comment. The tensor state unittest would fail without this `if`.